public/app/features/dashboard-scene/scene/DashboardDatasourceBehaviour.tsx

```diff
@@ -0,0 +1,59 @@
+import { SceneObjectBase, SceneObjectState, SceneQueryRunner, VizPanel } from '@grafana/scenes';
+import { SHARED_DASHBOARD_QUERY } from 'app/plugins/datasource/dashboard';
+
+import { findVizPanelByKey, getDashboardSceneFor, getQueryRunnerFor, getVizPanelKeyForPanelId } from '../utils/utils';
+
+import { DashboardScene } from './DashboardScene';
+
+interface DashboardDatasourceBehaviourState extends SceneObjectState {}
+
+export class DashboardDatasourceBehaviour extends SceneObjectBase<DashboardDatasourceBehaviourState> {
+  private prevRequestId: string | undefined;
+  public constructor(state: DashboardDatasourceBehaviourState) {
+    super(state);
+
+    this.addActivationHandler(() => this._activationHandler());
+  }
+
+  private _activationHandler() {
+    const queryRunner = this.parent;
+
+    if (!(queryRunner instanceof SceneQueryRunner)) {
+      throw new Error('DashboardDatasourceBehaviour must be attached to a SceneQueryRunner');
+    }
+
+    if (queryRunner.state.datasource?.uid !== SHARED_DASHBOARD_QUERY) {
```

Comment:
I'm also thinking that because of this line, searching for the correct query is pointless anyway, since the datasource will no longer be `Dashboard`, but `Mixed` with a `Dashboard` and a `Recorded query`.

Reply:
I think we can leave this as is now that we have a check in the dashboard `datasource.ts` file.